### DISCUSSION TITLE
DHT-11 problems

### DISCUSSION THREAD

**Discussion by toddc01b** [RavenSystem/esp-homekit-devices#1958] [General]
Hi all,

 I have a 3 wire DHT-11 (KY-015) connected to GPIO 2 but no matter what I do all it reports is 0% Humidity and 99 deg to home kit.

I've tried multiple sensors and different GPIO's but get the same result.

Am I missing something? Any help would be appreciated.

Thanks

**Comment by paulzhdanov** [comment]
Do you have pull up resistor between data pin and 3v3?

**Reply by toddc01b** [reply]
Yeh the 3 wire module has one integrated into it already.

**Reply by paulzhdanov** [reply]
so there are 2 possible problems: module is not operational OR config is wrong. No chance to help you with module, but you can post your config

**Reply by toddc01b** [reply]
I measured the resistor at approx 5K and it seems to be positioned correctly between Vcc and the data pin. I've also confirmed that the pins being used for 5v and gnd are good.
config is as below. I am trying to run it on the same module that is controlling the garage door.

{
  "c": {
    "o": 0
  },
  "a": [
    {
      "0": {
        "r": [
          {
            "g": 4,
            "v": 1,
            "i": 25
          }
        ]
      },
      "1": {
        "r": [
          {
            "g": 5,
            "v": 1,
            "i": 25
          }
        ]
      },
      "t": 40,
      "s": 5,
      "d": 30,
      "b": [
        {
          "g": 15,
          "t": 1
        }
      ]
    },
    {
      "t": 24,
      "b": [],
      "n": 1,
      "g": 2
    }
  ]
}

**Comment by paulzhdanov** [comment]
It looks ok except **"b": [],** try to get rid of that string and add **"j": 10,** instead (it's a polling time, frequency of how often to read the sensor). if that won't help, try another sensor, it may be physically broken

**Reply by toddc01b** [reply]
I'll try it and let you know the outcome. 
Thanks for your help so far.

**Reply by toddc01b** [reply]
So I tried the changes and, as expected, I can now see that the led on the module is flashing every 10 seconds as opposed to 30 seconds it was before. I'm still getting the same error with the temp and Rh though

I'm going to try using other GPIO's but I first need to solder pins to those so it might take me a while. I'll get back with the results when I have them.

Do you think it needs the "k" & "z" parameters? I wouldn't have though so but maybe I'll try and put those in also.

**Reply by toddc01b** [reply]
Hi,
I thought I would just provide an update.
I set in up on GPIO13 and it now seems to working fine. I did put the "k" & "z" parameters in during the same config change but I'm not sure if that made a difference or not.
Thanks for you help!!

**Reply by paulzhdanov** [reply]
You're welcome. Parameters k and z are just for calibration.

**Comment by Dikvadik** [comment]
Are you connecting the power and data wires to the module correctly?

**Reply by Dikvadik** [reply]
![image](https://user-images.githubusercontent.com/123756509/223549993-3b07a7f1-37a2-460e-b3f1-2722831cf7e6.jpeg)
